wxwidgets: fix building 3.1 on darwin

knl opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Description of changes

Fix build failures on darwin by passing all the necessary dependencies and removing deprecated command line switches to configure.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

tfmoraes commented:
@knl there is already a commit in Nixpkg to update wxwidgets https://github.com/NixOS/nixpkgs/pull/157848

knl commented:
@tfmoraes thanks, haven’t seen it, as I was not looking in staging. Once it reaches master, maybe this PR could be useful, since it fixes building on darwin?

Commit:
wxwidgets: fix building on darwin

Fix building on darwin by passing in all the dependencies and removing
compilation switches that do not exist anymore.

knl commented:
@tfmoraes I updated the PR to make 3.1 build on darwin. Please take a look. I'm wondering should we carve out a dedicated version without gtk, just for mac?

tfmoraes commented on pkgs/top-level/all-packages.nix:
```diff
@@ -20682,7 +20682,10 @@ with pkgs;
 
-  wxGTK31 = callPackage ../development/libraries/wxwidgets/wxGTK31.nix { };
+  wxGTK31 = callPackage ../development/libraries/wxwidgets/wxGTK31.nix {
+    withGtk2 = true;
```
it should be withGTK2 = false,  the idea is to compile with GTK 3.

knl replied:
The default in `wxGTK31.nix` is `withGtk2 ? true`. I changed it so that on darwin it's gtk3.

SuperSandro2000 commented on pkgs/development/libraries/wxwidgets/wxGTK31.nix:
```diff
@@ -17,20 +17,21 @@
 
 assert withMesa -> libGLU != null && libGL != null;
-assert withWebKit -> webkitgtk != null;
+assert withWebKit -> webkitgtk != null || stdenv.isDarwin;
```
```suggestion
assert withWebKit -> stdenv.isDarwin;
```

SuperSandro2000 commented on pkgs/development/libraries/wxwidgets/wxGTK31.nix:
```diff
@@ -17,20 +17,21 @@
 , darwin
 }:
 
 assert withMesa -> libGLU != null && libGL != null;
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/libraries/wxwidgets/wxGTK31.nix:
```diff
@@ -17,20 +17,21 @@
-, withWebKit ? false, webkitgtk
+, withWebKit ? stdenv.isDarwin
+, webkitgtk
 , darwin
```
```suggestion
, 
```
Please input the frameworks individually.

knl replied:
Gladly! I just took the existing style and was wondering why it's only having darwin.
